extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/GrpcServerRecorder.java

```diff
@@ -417,22 +424,57 @@ public RuntimeValue<ServerInterceptorStorage> initServerInterceptorStorage(
         return new RuntimeValue<>(new ServerInterceptorStorage(perServiceInterceptors, globalInterceptors));
     }
 
-    private Map.Entry<Integer, VertxServer> buildServer(Vertx vertx, GrpcServerConfiguration configuration,
+    @SuppressWarnings("rawtypes")
+    private Map.Entry<Integer, Server> buildServer(Vertx vertx, GrpcServerConfiguration configuration,
             Map<String, List<String>> blockingMethodsPerService, GrpcContainer grpcContainer, LaunchMode launchMode) {
+
+        Xds xds = configuration.xds;
+        boolean isXDS = xds != null && xds.enabled;
+
         int port = launchMode == LaunchMode.TEST ? configuration.testPort : configuration.port;
-        VertxServerBuilder builder = VertxServerBuilder.forAddress(vertx, configuration.host, port);
 
         AtomicBoolean usePlainText = new AtomicBoolean();
-        builder.useSsl(new Handler<HttpServerOptions>() { // NOSONAR
-            @Override
-            public void handle(HttpServerOptions options) {
+
+        ServerBuilder builder;
+        if (isXDS) {
+            ServerCredentials credentials = InsecureServerCredentials.create();
+            if (xds.secure) {
+                credentials = XdsServerCredentials.create(credentials);
+            }
+            builder = XdsServerBuilder.forPort(port, credentials);
+            // wrap with Vert.x context, so that the context interceptors work
+            VertxInternal vxi = (VertxInternal) vertx;
+            Executor delegate = vertx.nettyEventLoopGroup();
+            EventLoopContext context = vxi.createEventLoopContext();
+            Executor executor = command -> delegate.execute(() -> {
+                context.dispatch(command);
+            });
+            builder.executor(executor);
+            // custom XDS interceptors
+            if (launchMode == LaunchMode.DEVELOPMENT) {
+                builder.intercept(new DevModeInterceptor(Thread.currentThread().getContextClassLoader()));
+                // TODO -- see GrpcHotReplacementInterceptor usage -- OK?
```

Comment:
I think it should work ... it does look like Vert.x gRPC server's commandDecorator.
The question is why did you (or @michalszynkiewicz) go with the decorator in the first place?

OK, it's there to be used ... but imho you could get the same result with 2 interceptors as well
* DevInterptor -- instead of DevModeWrapper
* GrpcHotReplacementInterceptor -- and not have this in the command decorator consumer